app/src/main/java/com/github/factotum_sdp/factotum/data/localisation/Location.kt

```diff
@@ -53,45 +54,54 @@ class Location(query: String, context : Context) {
             if(!alreadyExists) cacheFile.appendText(toStore)
             return location
         }
+
+        fun geocoderQuery(query: String, context: Context): List<Address>?{
+            // must handle differently depending on SDK.
+            // getLocationFromName(String, int) deprecated in SDK 33
+            val geocoder = Geocoder(context)
+            return if (Build.VERSION.SDK_INT >= TIRAMISU) {
+                tiramisuResultHandler(query, geocoder)
+            } else{
+                resultHandler(query, geocoder)
+            }
+        }
+
+        @RequiresApi(TIRAMISU)
+        private fun tiramisuResultHandler(query: String, geocoder: Geocoder): List<Address>? {
+            var result : List<Address>? = listOf()
+            val latch = CountDownLatch(1)
```

Comment:
So maybe here instead of using a CountDownlatch that will block the main threads and could be the reason why you can't display new result when adding/deleting letter in the search bar, you could use Coroutines that will be using other threads, like in this case maybe the one for IO?